node/engine/engine.go

```diff
@@ -61,9 +65,10 @@ type Engine struct {
 	ObjectiveRequestsFromAPI chan protocols.ObjectiveRequest
 	PaymentRequestsFromAPI   chan PaymentRequest
 
-	fromChain  <-chan chainservice.Event
-	fromMsg    <-chan protocols.Message
-	fromLedger chan consensus_channel.Proposal
+	fromChain    <-chan chainservice.Event
+	fromMsg      <-chan protocols.Message
+	fromLedger   chan consensus_channel.Proposal
+	signRequests <-chan p2pms.SignatureRequest
```

Comment:
Do you think we will be able to reuse this interface to solve these issues:
https://github.com/statechannels/go-nitro/issues/1508
#1075

Reply:
### Issue 1508
We will likely need to add another config option for the `libp2p` peerId private key (which is used to derive the peerId). We need to pass that private key to the `libp2p` config options in the following code:
```
	options := []libp2p.Option{
		libp2p.Identity(privateKey),
		...
	}
```
Right now the `messageservice` holds that private key in-memory. To improve consistency, we could put that private key in the `store` (as we do with the scAddr). We could then update the `engine.handleSignRequests` method to make both signatures on the DHT record:
1. Sign the record using the scAddr private key
2. Sign the record using the peerId private key
3. Return both signatures to the `messageservice` so that the record can be published to the DHT

### Issue 1075
I'm not sure how to resolve this issue. It seems that `libp2p` requires the peerId private key during its configuration, and the key is used within `libp2p` abstractions to sign DHT records and encrypt p2p communication. By using separate keys for the peerId and scAddr (as discussed above in issue 1508), we would at least mitigate risk by reducing responsibilities/purposes of each key